pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -794,14 +788,15 @@ contract StablePhantomPool is StablePool, ProtocolFeeCache {
         view
         returns (uint256 virtualSupply, uint256[] memory amountsWithoutBpt)
     {
-        // The initial amount of BPT pre-minted is _MAX_TOKEN_BALANCE and it goes entirely to the pool balance in the
-        // vault. So the virtualSupply (the actual supply in circulation) is defined as:
-        // virtualSupply = totalSupply() - (_balances[_bptIndex] - _dueProtocolFeeBptAmount)
+        // In normal operation, no BPT are minted or burned, so we *could* use `_MAX_TOKEN_BALANCE` instead of
+        // `totalSupply()` and save gas:
+        // virtualSupply = _MAX_TOKEN_BALANCE - amounts[_bptIndex] + _dueProtocolFeeBptAmount;
         //
-        // However, since this Pool never mints or burns BPT outside of the initial supply (except in the event of an
-        // emergency pause), we can simply use `_MAX_TOKEN_BALANCE` instead of `totalSupply()` and save
-        // gas.
-        virtualSupply = _MAX_TOKEN_BALANCE - amounts[_bptIndex] + _dueProtocolFeeBptAmount;
+        // However, while the Pool is in recovery mode, LPs can exit proportionally and burn BPT. Since recovery
+        // mode is reversible - and unlike pausing, allows normal operation while enabled, including swaps -
+        // we cannot use this optimization. Otherwise, burning BPT in recovery mode would make the accounting
+        // permanently inaccurate (and the pool unsafe).
```

Comment:
I think this will create more confusion than anything. I don't think we need to explain why we are not making a hypothetical optimization that'd be available of the system worked differently. The fact that this used to be possible is irrelevant.

I'd delete these two paragraphs.